docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+    self.provider.io.error("An error occured during token translation :( . Please see PythonSdk log for details")
+```
+
+`model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})`
+
+Here we load a model that(soon) will contain a `TextVectorization` TensorFlow layer at `model.layers[0]` where `translational` is a comma-separated value of integer token values to be "translated" to their original string representations as mapped by said `model.layers[0]` layer.
+
+
+For now, we won't have any `translationVal`s, so this won't be relevant until later sections.
+Then, we log in to `self.info` as defined below:
+
+#### Function: `self.get_token_translation`
+
+```python
+def info(self, s: str):
+  """Log s, and then send an `info` message to designer containg s"""
+  logger.info(s)
+  self.provider.io.info(s)
+```
+
+While the method is relatively straightforward, it includes(spellcheck) a notable point: there is a distinct difference between **logging** and **messaging**.
+We chose this for many reasons, a large one being that the tool _User_ rarely needs, and even less so *wants*, to see the debugging output.
+Do not use the above as a crunch or without consideration; logging is cheap, but _messaging_ is expensive!
+A good rule of thumb is to keep your `self.provider.io` calls to a minimum, only sending what your USER needs to see.
+You can leverage `logging` more freely, especially when paired with the available debugging flag (NOTE: reference needed here).
+Use sparingly, or you will risk slowing your tool with expensive IO! _(A very important consideration in ML)_
+
+Finally, update `on_complete` to call our `send_preview_data()` function:
+
+```python
+self.provider = provider
+# ...
+self.info("Plugin initialized.")
+if self.MODE == 'DATA':
+  # ...
+elif self.MODE == 'PREVIEW':
+  self.send_preview_data()
+```
+
+Now, that we have data and a way to feed it as input into our model we are ready to move on to the next mode.
+
+### `TRAIN` mode
+
+Now so far, we've:
+* Created custom methods to support TensorFlow API calls
+* Added `DATA` mode, a way for a user to input and load a data source as a dataset
+* Added `PREVIEW` mode, a way for a user to request preview data from the backend for feedback during data prepping
+* Mocked out (actual tests needed) our frontend UI object and tested to verify functionality
+  
+In our third mode, `MODEL`, a user(NOTE: change to this phrasing in earlier paragraphs?) will input:
+
+* Where to store their generated and trained text classifier NN
+* What to name the model
+* Various tunable model training parameters
+* a TextVectorization option
+
+Then, we will collect the reporting data TensorFlow provides for training and send it to the UI SDK (frontend) for the User to see dynamically generated visual feedback and data from the training session(s).
```

Comment:
TW review:

In the third mode, `MODEL`, we input...

* Where to store the generated and trained Text Classifier NN.
* What to name the model.
* Various tunable model training parameters.
* A TextVectorization option.

Then, we collect the reporting data TensorFlow provides for training and send it to the UI SDK (front end) for the user to see dynamically generated visual feedback and data from the training sessions.